### DISCUSSION TITLE
Difference between this and importing straight into Photoprism?

### DISCUSSION THREAD

**Discussion by thibaultmol** [TheLastGimbus/GooglePhotosTakeoutHelper#211] [General]
I'm not trying to downplay your project, I'm glad it exists.

But I was wondering for people switching from Gphotos to Photoprism: their auto import system seems to get the correct dates and stuff from the json files as well. Only thing it doesn't do and needs a 3rdparty script for atm is to get albums correctly.

So in that case you wouldn't need GPTH then?
https://docs.photoprism.app/user-guide/use-cases/google/

**Comment by TheLastGimbus** [comment]
I didn't bite deep into this, but here is my perdicion:

photoprism might be better about metadata stuff, since gpth currently doesn't merge anything of this into exif (geolocation etc)

BUT, they may not have a 3-year-old expirience into finding correct dates (look at https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/tree/master/lib/date_extractors - this ain't just `photoName + ".json"`) - where json names are stripped to 51 characters, `-modified` suffixes, etc etc. Even after 3 years, i still have issues like #204 ... 

gpth has a lot of fallbacks implemented like file name guessing - just to get the date now

last - as they noted, they do not support albums - gpth finally does :tada:

tl;dr i think both ways will be good if you don't care that much about modified locations or smth

**Reply by thibaultmol** [reply]
Cool, perfect! 

Very good arguments :+1:
